### ISSUE TITLE
[Bug]: tealdeer config file should not be in `%USERPROFILE%/scoop/apps/tealdeer/current`

### ISSUE THREAD

**Issue by nagromc** [ScoopInstaller/Main#5500]
### Prerequisites

- [X] I have written a descriptive issue title.
- [X] I have searched all issues/PRs to ensure it has not already been reported or fixed.
- [X] I have verified that I am using the latest version of Scoop and corresponding bucket.

### Package Name

tealdeer

### Expected/Current Behaviour

I was trying to change the default configuration, but I realized the config file is located in `%USERPROFILE%/scoop/apps/tealdeer/current/` instead of `%USERPROFILE%/AppData/Roaming/tealdeer/tealdeer/`.

The problem with this approach is that:
- The configuration will not be migrated when the app is updated
- The users that maintain a dotfile repository expect to find config files in `%USERPROFILE%/AppData`

### Steps to Reproduce

```console
tldr --show-paths
Config dir:       C:\Users\foo\scoop\apps\tealdeer\current\ (env variable)
Config path:      C:\Users\foo\scoop\apps\tealdeer\current\config.toml
Cache dir:        C:\Users\foo\scoop\persist\tealdeer\cache (config file)
Pages dir:        C:\Users\foo\scoop\persist\tealdeer\cache\tldr-pages\
Custom pages dir: C:\Users\foo\scoop\persist\tealdeer\custom-pages (config file)
```

### Possible Solution

Tealdeer offers several environment variables that can be used to solve this problem.

### Scoop and Buckets Version

```console
Current Scoop version:
v0.3.1 - Released at 2022-11-15

'extras' bucket:
21c454e90 (HEAD -> master, origin/master, origin/HEAD) yuescript: Update to version 0.21.8

'games' bucket:
94e13e7d4 (HEAD -> master, origin/master, origin/HEAD) duckstation-preview: Update to version 20240205-geafa5d4

'java' bucket:
4bf5794c (HEAD -> master, origin/master, origin/HEAD) intellij-jbr17: Update to version 17.0.9-b1087.11

'main' bucket:
2b64225dd (HEAD -> master, origin/master, origin/HEAD) stgit: Update to version 2.4.3

'morgan' bucket:
5a052ca (HEAD -> master, origin/master, origin/HEAD) hoppscotch: add version 23.12.1-1

'nerd-fonts' bucket:
8d9b6f9 (HEAD -> master, origin/master, origin/HEAD) smiley-sans-dev: Update to version 20240203.c9a011d

'nonportable' bucket:
f9709f5 (HEAD -> master, origin/master, origin/HEAD) 10x-np: Update to version 1.0.125

'versions' bucket:
45952f510 (HEAD -> master, origin/master, origin/HEAD) zig-dev: Update to version 0.12.0-dev.2586
```

### Scoop Config

```console
aria2-warning-enabled : False
last_update           : 05/02/2024 09:58:29
use_external_7zip     : True
scoop_branch          : master
scoop_repo            : https://github.com/ScoopInstaller/Scoop
```

### PowerShell Version

```console
Name                           Value
----                           -----
PSVersion                      7.4.1
PSEdition                      Core
GitCommitId                    7.4.1
OS                             Microsoft Windows 10.0.19044
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by nagromc** [comment]
I can submit a PR if you wish.

**Comment by nagromc** [comment]
Alright; I’ll push it soon.

**Comment by SorYoshino** [comment]
> * The configuration will not be migrated when the app is updated

User configurations will not be lost, as the relevant files or folders have been stored in `$persist_dir`.

> * The users that maintain a dotfile repository expect to find config files in `%USERPROFILE%/AppData`

Frankly speaking, rather than having the application store its configuration data under `$env:APPDATA`, I would prefer to see it stored in `$persist_dir`. When it is necessary to back up all configuration related to the application, operating directly on the entire `$persist_dir` folder is far more convenient than searching for items one by one under `$env:APPDATA`.

I believe that the vast majority of Scoop users prefer relevant data to be stored in `$persist_dir`, and will look for configuration there first. When persistence of the associated configuration is functioning properly, this behavior should not be changed.

**Comment by Lutra-Fs** [comment]
See also https://github.com/ScoopInstaller/Main/issues/5013
